src/physics/navier_stokes.cpp

```diff
@@ -565,6 +565,55 @@ ::dissipative_flux_templated (
     return viscous_flux;
 }
 
+template <int dim, int nstate, typename real>
+void NavierStokes<dim,nstate,real>
+::boundary_no_slip_wall (
+   const dealii::Tensor<1,dim,real> &normal_int,
+   const std::array<real,nstate> &soln_int,
+   const std::array<dealii::Tensor<1,dim,real>,nstate> &soln_grad_int,
+   std::array<real,nstate> &soln_bc,
+   std::array<dealii::Tensor<1,dim,real>,nstate> &soln_grad_bc) const
+{
+    // Slip wall boundary conditions (No penetration)
+    // Given by Algorithm II of the following paper
+    // Krivodonova, L., and Berger, M.,
+    // “High-order accurate implementation of solid wall boundary conditions in curved geometries,”
+    // Journal of Computational Physics, vol. 211, 2006, pp. 492–512.
+    const std::array<real,nstate> primitive_interior_values = convert_conservative_to_primitive<real>(soln_int);
+
+    // Hard-coded for isothermal boundary
+    const double temperature_wall = this->temperature_inf; // add a parameter for this
+
+    // Copy density and pressure
+    std::array<real,nstate> primitive_boundary_values;
+    primitive_boundary_values[0] = primitive_interior_values[0];
+    primitive_boundary_values[nstate-1] = this->compute_pressure_from_density_temperature(primitive_boundary_values[0], temperature_wall);
```

Comment:
We could include here an option for isothermal and adiabatic. In the case of adiabatic you would just set the pressure at the boundary to the pressure in the interior value.

Reply:
Thanks for letting me know, I've addressed this in the most recent commit.